File: js/avoidObstacles.js
function avoidObstacles(inputArray) {
  var i = 2;
  while(true){
    if(inputArray.every(x => x%i != 0)){
      return i;
    }
    i++;
  }
}

console.log(avoidObstacles([1,5,7,8]))

File: python/avoidObstacles.py
def avoidObstacles(inputArray):
    maxnum = max(inputArray)
    for i in range(maxnum)[2:] :
        if all([j%i != 0 for j in inputArray]) :
            return i
    return maxnum + 1




print avoidObstacles([1,4,7,8,9])
